src/main/java/entity/Ingredient.java

```diff
@@ -32,4 +32,11 @@ public String getName() {
     public void setName(String name) {
         this.name = name;
     }
+
+    public boolean equals(Ingredient i){
```

Comment:
I think this should be marked `@Override` as well

Reply:
I'm a little confused. In my opinion, they are not subclass and parent classes so they don't need to mark Override, is that correct? Though they have the same name, they are not replacing each other, right?

Reply:
They're overriding `Class`'s default `equals` method.